### PR TITLE
Fix worker are notified to stop with non_graceful shutdown

### PR DESCRIPTION
**Pull request by fakeshadow**
## PR Type

Bug Fix

## PR Checklist
Check your PR fulfills the following:

- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview

With non graceful shutdown there is no signal for notify worker threads to exit. This could result in leak of worker threads.

This PR changes to notify worker threads to shutdown on all cases. But only in graceful shutdown case server builer wait for worker shutdown response.

### CODE REVIEW

**Comment by fakeshadow** [comment]
This changes the behavior of server as force shutdown was not exiting worker threads. There could be people using it as a feature rather than bug.

File: actix-server/src/builder.rs [hunk 1]
```diff
@@ -381,45 +381,29 @@ impl ServerBuilder {
...
                         let _ = tx.send(());
                     }
                     for tx in notify {
                         let _ = tx.send(());
                     }
-                }
+
+                    if exit {
+                        sleep(Duration::from_millis(300)).await;
+                        System::current().stop();
```
Comment (robjtede) [inline actix-server/src/builder.rs:404]: should we code this as try_current ?

Comment (fakeshadow) [reply actix-server/src/builder.rs:404]: Right now we still expect system when running a server so I think it's fine here.
We can change when actix-rt is made optional.

Comment (fakeshadow) [reply actix-server/src/builder.rs:404]: The thing I have no idea is why there is 300 ms delay.

**Review by robjtede — APPROVED** [review]

**Pull request merged**